test/functional/lua/comment_spec.lua

```diff
@@ -0,0 +1,644 @@
+local helpers = require('test.functional.helpers')(after_each)
+
+local api = helpers.api
+local clear = helpers.clear
+local command = helpers.command
+local eq = helpers.eq
+local exec_capture = helpers.exec_capture
+local exec_lua = helpers.exec_lua
+local feed = helpers.feed
+
+-- Reference text
+-- aa
+--  aa
+--   aa
+--
+--   aa
+--  aa
+-- aa
+local example_lines = { 'aa', ' aa', '  aa', '', '  aa', ' aa', 'aa' }
+
+local set_commentstring = function(commentstring)
+  api.nvim_set_option_value('commentstring', commentstring, { buf = 0 })
+end
+
+local get_lines = function(from, to)
+  from, to = from or 0, to or -1
+  return api.nvim_buf_get_lines(0, from, to, false)
+end
+
+local set_lines = function(lines, from, to)
+  from, to = from or 0, to or -1
+  api.nvim_buf_set_lines(0, from, to, false, lines)
+end
+
+local set_cursor = function(row, col)
+  api.nvim_win_set_cursor(0, { row, col })
+end
+
+local get_cursor = function()
+  return api.nvim_win_get_cursor(0)
+end
+
+local setup_treesitter = function()
+  command('filetype plugin on')
+  -- NOTE: This leverages bundled Vimscript and Lua tree-sitter parsers
+  api.nvim_set_option_value('filetype', 'vim', { buf = 0 })
+  exec_lua('vim.treesitter.start()')
+end
+
+before_each(clear)
+
+describe('commenting', function()
+  before_each(function()
+    set_lines(example_lines)
+    set_commentstring('# %s')
+    command('source $VIMRUNTIME/lua/vim/_defaults.lua')
```

Comment:
```suggestion
  -- NOTE: This leverages bundled Vimscript and Lua tree-sitter parsers
  api.nvim_set_option_value('filetype', 'vim', { buf = 0 })
  exec_lua('vim.treesitter.start()')
end

before_each(function()
  clear({ args_rm = { '--cmd' }, args = { '--clean' } })
end)

describe('commenting', function()
  before_each(function()
    set_lines(example_lines)
    set_commentstring('# %s')
```

Reply:
Thanks! I was struggling with proper initialization of child process and this was the best solution I could find.